src/ECER.Clients.RegistryPortal/ECER.Clients.RegistryPortal.Server/Communications/CommunicationsEndpoints.cs

```diff
@@ -0,0 +1,54 @@
+﻿using ECER.Managers.Registry.Contract.Communications;
+using ECER.Utilities.Hosting;
+using ECER.Utilities.Security;
+using Wolverine;
+
+namespace ECER.Clients.RegistryPortal.Server.Communications;
+
+public class CommunicationsEndpoints : IRegisterEndpoints
+{
+  public void Register(IEndpointRouteBuilder endpointRouteBuilder)
+  {
+    endpointRouteBuilder.MapGet("/api/messages", async (HttpContext httpContext, IMessageBus messageBus) =>
+    {
+      var userContext = httpContext.User.GetUserContext();
+      var query = new CommunicationsQuery();
+      query.ByIdentity = userContext!.Identity!;
+      query.ByStatus = new CommunicationStatus[] { CommunicationStatus.NotifiedRecipient, CommunicationStatus.Acknowledged };
+      var results = await messageBus.InvokeAsync<CommunicationsQueryResults>(query);
+
+      return TypedResults.Ok(results.Items.Select(i => new Communication
+      {
+        Id = i.Id!,
+        Subject = i.Subject,
+        Text = i.Text
+      }));
+    })
+     .WithOpenApi("Handles messages queries", string.Empty, "message_get")
+     .RequireAuthorization();
+
+
+    endpointRouteBuilder.MapGet("/api/messages/status", async (HttpContext httpContext, IMessageBus messageBus) =>
```

Comment:
Not a very restul uri. I think it is worth considering moving the message status to to the userinfo endpoint - if a user if found, return the aggregated message number.

Reply:
we talked about it and decided to keep it as it is for now